cmake/checks/check_02_compiler_features.cmake

```diff
@@ -501,3 +501,44 @@ IF(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
 
   RESET_CMAKE_REQUIRED()
 ENDIF()
+
+
+#
+# CCache support is only really useful for developers. 
+# We therefore disable the feature by default.
+#
+# See https://invent.kde.org/kde/konsole/-/merge_requests/26/diffs
+# and https://stackoverflow.com/a/24305849
+#
+SET(BUILD_USING_CCACHE FALSE CACHE BOOL "Enable ccache support")
+
+MESSAGE(STATUS "")
+MESSAGE(STATUS "Looking for CCache.")
+IF(BUILD_USING_CCACHE)
+  #
+  # Built-in CCache support requires CMake version 2.8 or newer
+  #
+  IF(CMAKE_VERSION VERSION_LESS 2.8)
+    SET(BUILD_USING_CCACHE FALSE)
+
+    MESSAGE(STATUS "deal.II requires CMake version 2.8, or newer for CCache support.\n"
+                   "Reconfigure with a sufficient cmake version.")
+  ELSE()
+    FIND_PROGRAM(CCACHE_FOUND "ccache")
+
+    IF (CCACHE_FOUND)
```

Comment:
Also, I am not a huge fan of searching for a ccache executable (that never gets used).